internal/misc/fsolve.go

```diff
@@ -0,0 +1,170 @@
+// Derived from SciPy's special/c_misc/fsolve.c
```

Comment:
Yeah, I was meaning to ask what you wanted to do about that. I was just mirroring the directory structure, but `c_misc` seemed misleading. What would you suggest?